collector/nic-module.go

```
@@ -636,14 +652,28 @@ func discoverMellanoxDevices() ([]DeviceInfo, error) {
 }
 
 func (n *NicModuleCollector) runMlxlink(hostname string, systemserial string, slot string, device DeviceInfo, resp chan runMlxlinkResponse) {
-	cmd := exec.Command("mlxlink", "-json", "-d", device.pciAddress, "-m", "-c")
+	cmd := exec.Command("mlxlink", "-json", "-d", device.pciAddress, "-m", "-c", "--rx_fec_histogram", "--show_histogram")
 	output, err := cmd.CombinedOutput()
+	mlxout := gjson.Parse(string(output))
+	valid_output := false
 	if err == nil {
-		metrics := parseOutput(string(output), hostname, systemserial, slot, device)
-		resp <- runMlxlinkResponse{metrics, false}
-
+		valid_output = true
 	} else {
+		// Check and see if there was an error relating to the FEC histogram param - if there is, we can still provide all
+		//  of the other metrics
+		if mlxout.Get("result.output.Operational Info.State").String() == "Active" {
```

I thought it was when it was not active!

---

I'd probably ignore err if we can parse the result and just check the error code.

---

Other errors might return 1